src/core/net/nat64_translator.cpp

```diff
@@ -510,6 +510,8 @@ Error Translator::SetIp4Cidr(const Ip4::Cidr &aCidr)
     return err;
 }
 
+void Translator::ClearIp4Cidr(void) { mIp4Cidr.Clear(); }
```

Comment:
I would suggest aldo do the following things in this function:

```
    mAddressMappingPool.FreeAll();
    mActiveAddressMappings.Clear();
    mIp4AddressPool.Clear();

    UpdateState();
    Get<Notifier>().Signal(kEventNat64TranslatorStateChanged);
```

Reply:
Good catch!